### PR TITLE
Group references

### PR DESCRIPTION
**Pull request by stefanopuddu**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/com/agiletec/aps/system/services/authorization/AuthorizationService.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Override
+    public List<UserDto> getGroupUtilizer(String groupCode) {
+        try {
+
+            List<String> usernames = ((GroupUtilizer<String>) this.getAuthorizationManager()).getGroupUtilizers(groupCode);
+            List<UserDto> dtoList = new ArrayList<>();
+            if (null != usernames) {
+                usernames.parallelStream().forEach(i -> {
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/authorization/AuthorizationService.java]: For parallel stream, lets only use when we know its needed. For small workloads a sequential stream should be faster.

File: engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/ContentService.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        return ((IManager) this.getContentManager()).getName();
+    }
+
+    @Override
+    public List<ContentDto> getGroupUtilizer(String groupCode) {
+        try {
+            List<String> resourcesId = ((GroupUtilizer<String>) this.getContentManager()).getGroupUtilizers(groupCode);
+            List<ContentDto> dtoList = new ArrayList<>();
+            if (null != resourcesId) {
+                resourcesId.parallelStream().forEach(i -> {
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/ContentService.java]: Same as above.

File: engine/src/main/java/org/entando/entando/aps/system/services/dataobject/DataObjectService.java [hunk 3]
```diff
@@ -138,4 +176,34 @@ public void setDataObjectModelManager(IDataObjectModelManager dataObjectModelMan
...
+
+    @Override
+    public List<DataObjectDto> getGroupUtilizer(String groupCode) {
+        try {
+            boolean onLine = true;
+            DataObjectManager entityManager = (DataObjectManager) this.extractEntityManager(getManagerName());
+            List<String> idList = ((GroupUtilizer<String>) entityManager).getGroupUtilizers(groupCode);
+            List<DataObjectDto> dtoList = new ArrayList<>();
+            if (null != idList) {
+                idList.parallelStream().forEach(i -> {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/dataobject/DataObjectService.java]: Same as above.

File: engine/src/main/java/org/entando/entando/aps/system/services/dataobject/model/DataObjectDto.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+public class DataObjectDto {
+
+    private String id;
+    private String typeCode;
+    private String typeDescription;
+    private String description;
+    private String mainGroup;
+    private Set<String> groups;
+    //private List<AttributeInterface> _attributeList;
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/dataobject/model/DataObjectDto.java]: Remove

File: engine/src/main/java/org/entando/entando/aps/system/services/group/GroupService.java [hunk 5]
```diff
@@ -215,4 +239,43 @@ protected BeanPropertyBindingResult checkGroupForDelete(Group group) throws ApsS
...
+        Optional<GroupServiceUtilizer> defName = beans.values().stream()
+                                               .filter(service -> service.getManagerName().equals(managerName))
+                                                      .findFirst();
+        if (defName.isPresent()) {
+            return defName.get();
+        }
+        return null;
+    }
+
+    //    public Map<String, List<UtilizerEntry>> getReferencingObjects(Group group) {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/group/GroupService.java]: Remove

File: engine/src/main/java/org/entando/entando/aps/system/services/group/model/GroupDto.java [hunk 6]
```diff
@@ -72,11 +70,11 @@ public static String getEntityFieldName(String dtoFieldName) {
         }
     }
 
-    public Map<String, List<UtilizerEntry>> getReferences() {
+    public Map<String, Boolean> getReferences() {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/group/model/GroupDto.java]: Pretty generic. Add comment on contents

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**